### PR TITLE
Architectural refactors to pave the way for ModelBaseline to evaluate a model's generated outputs

### PR DESCRIPTION
**Pull request by klxu03**
Can now test the outputs generated from a model in src/evaluate_model.py

Changed the bounds for relevant HTML context to 15 above, 30 below. The upper and lower bounds are stored in their respective variables. Upper means lines above, lower means lines below 

Fixed TAP tasks 19th partition not properly testing random baseline

Added Poetry support

### CODE REVIEW

**Comment by klxu03** [comment]
@danyaljj

**Review by danyaljj — APPROVED** [review]

File: poetry.lock [hunk 1]
```diff
@@ -0,0 +1,1365 @@
+# This file is automatically @generated by Poetry 1.6.1 and should not be changed by hand.
```
Comment (danyaljj) [inline poetry.lock:1]: drop?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+[tool.poetry]
```
Comment (danyaljj) [inline pyproject.toml:1]: drop?

Comment (klxu03) [reply pyproject.toml:1]: Should be kept since this is a poetry config file for people using the Poetry file manager

**Pull request merged**